### ISSUE TITLE
Refactor scheduler and maintenance test to envtest

### ISSUE THREAD

**Issue by timebertt** [gardener/gardener#4655]
**How to categorize this issue?**

/area testing
/kind technical-debt

## TL;DR

Currently, both the [scheduler test](https://github.com/gardener/gardener/blob/712a875b5965c9c87a8ce9c304c13733fe4b11ce/test/integration/gardener/scheduler/scheduler_test.go#L93) and the [maintenance test](https://github.com/gardener/gardener/blob/ded9a1dcc9631309e283c8eacfa9c693e31eaee4/test/integration/shoots/maintenance/maintenance.go#L117) are not executed.

## Explanation

There are several conceptual problems with these tests:
- the scheduler test scales down gardener-controller-manager
  - this is from old times before the gardenlet and the motiviation is to prevent shoots from actually being created
- the maintenance test scales down gardener-scheduler
  - this is also to prevent shoots from being eventually created on a real seed
- the scale down operations cannot be done in a local setup (as both components run as processes on the dev machine instead of pods)
- the scale down operations cannot be done if a virtual-garden setup is used (i.e. the components run on a different hosting cluster and cannot be scaled down by using the garden client) 
- the maintenance test manipulates whatever `CloudProfile` the configured test shoot uses

To summarize: these tests are actually "integration tests" but are built as (test machinery style) "e2e tests".

The result is:
- we don't want to run these tests against a real landscape (not even a development landscape), so it's not executed in the test machinery for release validation or release tests
- nobody runs these tests manually against a local setup
- tests are not executed as part of PR verification
- tests have been unmaintained for a while

## Proposal

We should refactor both tests to "real" integration tests leveraging envtest (see https://github.com/gardener/gardener/issues/3765).
The tests don't need real landscape or any external infrastructure and just deal with some API objects, meaning they can simply be executed against a dummy control plane.

The benefits are:
- tests are more lightweight and run much faster
- tests can be executed for every PR verification, release job, etc.
- tests can be executed locally with ease

**Comment by ialidzhikov** [comment]
/assign @acumino
for the maintenance test refactoring
